cider/function/FunctionLookupEngine.cpp

```diff
@@ -195,15 +283,101 @@ const FunctionDescriptor FunctionLookupEngine::lookupFunction(
             from_platform,
             from_platform_));
   }
-  function_descriptor.func_sig = function_signature;
-  function_descriptor.op_support_expr_type = getFunctionOpSupportType(function_signature);
-  if (function_descriptor.op_support_expr_type == OpSupportExprType::kFUNCTION_OPER) {
+  const std::string& function_name = function_signature.func_name;
+  auto real_function_name = getRealFunctionName(function_name);
+  FunctionSignature function_signature_result = function_signature;
+  function_signature_result.func_name = real_function_name;
+  function_descriptor.func_sig = function_signature_result;
+  auto funtion_op_support_type_result = getFunctionOpSupportType(function_signature);
+  function_descriptor.op_support_expr_type = funtion_op_support_type_result;
+  if (funtion_op_support_type_result != OpSupportExprType::kUNDEFINED_EXPR) {
+    function_descriptor.is_cider_support_function = true;
+  }
+  // extension function, no need to look up internal scalar and agg
+  if (funtion_op_support_type_result == OpSupportExprType::kFUNCTION_OPER) {
+    return function_descriptor;
+  }
+  auto funtion_scalar_op_result = getFunctionScalarOp(function_signature);
+  function_descriptor.scalar_op_type = funtion_scalar_op_result;
+  if (funtion_scalar_op_result != SQLOps::kUNDEFINED_OP) {
+    function_descriptor.is_cider_support_function = true;
     return function_descriptor;
   }
-  function_descriptor.scalar_op_type = getFunctionScalarOp(function_signature);
-  if (function_descriptor.scalar_op_type != SQLOps::kUNDEFINED_OP) {
+  auto funtion_string_op_result = getFunctionStringOp(function_signature);
+  function_descriptor.string_op_type = funtion_string_op_result;
+  if (funtion_string_op_result != SqlStringOpKind::kUNDEFINED_STRING_OP) {
+    function_descriptor.is_cider_support_function = true;
     return function_descriptor;
   }
-  function_descriptor.agg_op_type = getFunctionAggOp(function_signature);
+  auto funtion_agg_op_result = getFunctionAggOp(function_signature);
+  function_descriptor.agg_op_type = funtion_agg_op_result;
+  if (funtion_agg_op_result != SQLAgg::kUNDEFINED_AGG) {
+    function_descriptor.is_cider_support_function = true;
+  }
+  return function_descriptor;
+}
+
+const FunctionDescriptor FunctionLookupEngine::lookupFunction(
+    const std::string& function_signature_str,
+    const std::string& function_return_type_str,
+    const PlatformType& from_platform) const {
+  FunctionDescriptor function_descriptor;
+  auto function_name =
+      function_signature_str.substr(0, function_signature_str.find_first_of(':'));
+  auto function_args = function_signature_str.substr(
```

Comment:
So here we rely on the correctness of function args types defined in substrait plan. Will it be complicated if we get the exact arg types from the expression?

Reply:
Yes, it will be more complicated. So we should timely update the yaml file when update and add the cider function.